homeassistant/components/remote_rpi_gpio/binary_sensor.py

```diff
@@ -0,0 +1,104 @@
+"""Support for binary sensor using RPi GPIO."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import remote_rpi_gpio
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ADDRESS = 'address'
+CONF_BOUNCETIME = 'bouncetime'
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_PORTS = 'ports'
+CONF_PULL_MODE = 'pull_mode'
+
+DEFAULT_BOUNCETIME = 50
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_PULL_MODE = "UP"
+
+DEPENDENCIES = ['remote_rpi_gpio']
+
+_SENSORS_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADDRESS): cv.string,
+    vol.Required(CONF_PORTS): _SENSORS_SCHEMA,
+    vol.Optional(CONF_BOUNCETIME, default=DEFAULT_BOUNCETIME): cv.positive_int,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    vol.Optional(CONF_PULL_MODE, default=DEFAULT_PULL_MODE): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Raspberry PI GPIO devices."""
+    address = config.get(CONF_ADDRESS)
+    pull_mode = config.get(CONF_PULL_MODE)
+    bouncetime = config.get(CONF_BOUNCETIME)
+    invert_logic = config.get(CONF_INVERT_LOGIC)
+
+    binary_sensors = []
+    ports = config.get('ports')
+    for port_num, port_name in ports.items():
+        binary_sensors.append(RemoteRPiGPIOBinarySensor(
+            port_name, address, port_num, pull_mode, bouncetime, invert_logic))
+    add_entities(binary_sensors, False)
+
+
+class RemoteRPiGPIOBinarySensor(BinarySensorDevice):
+    """Represent a binary sensor that uses a Remote Raspberry Pi GPIO."""
+
+    def __init__(self, name, address, port, pull_mode,
+                 bouncetime, invert_logic):
+        """Initialize the RPi binary sensor."""
+        self._name = name or DEVICE_DEFAULT_NAME
+        self._address = address
+        self._port = port
+        self._pull_mode = pull_mode
+        self._bouncetime = bouncetime
+        self._invert_logic = invert_logic
+        self._state = None
+        self._button = None
+
+        try:
+            self._button = remote_rpi_gpio.setup_input(self._address,
+                                                       self._port,
+                                                       self._pull_mode)
+        except (ValueError, IndexError, KeyError):
+            return None
+
+        def read_gpio():
+            """Read state from GPIO."""
+            self._state = remote_rpi_gpio.read_input(self._button)
+            self.schedule_update_ha_state()
+
+        self._state = remote_rpi_gpio.read_input(self._button)
+
+        self._button.when_released = read_gpio
```

Comment:
Move registration of update callbacks to entity coroutine method `async_added_to_hass`.